core/options.c

```diff
@@ -104,6 +104,64 @@ static void ignore_varargs_function(char *format, ...) { }
 
 #endif /* NOT_DYNAMORIO_CORE */
 
+#include <stddef.h>
+
+typedef enum option_type_t {
+    OPTION_TYPE_bool,
+    OPTION_TYPE_uint,
+    OPTION_TYPE_uint_addr,
+    OPTION_TYPE_uint_size,
+    OPTION_TYPE_uint_time,
+    OPTION_TYPE_pathstring_t,
+    OPTION_TYPE_liststring_t,
+} option_type_t;
+
+/* Function for identifying string type. */
+bool
+is_string_type(enum option_type_t type)
+{
+    return type == OPTION_TYPE_pathstring_t || type == OPTION_TYPE_liststring_t;
+}
+
+typedef enum option_modifier_t {
+    OPTION_MOD_STATIC,
+    OPTION_MOD_DYNAMIC
+} option_modifier_t;
+
+typedef uint uint_size;
+typedef uint uint_time;
+typedef ptr_uint_t uint_addr;
+
+/* Structure with all the information about an option. */
+typedef struct _option_t {
+    const char *name;
+    uint64 offset;
+    uint64 size;
+    option_type_t type;
+    op_pcache_t affects_pcache;
+    option_modifier_t modifier;
+} option_t;
+
+/* Create an array with traits of all the internal options. */
+#ifndef EXPOSE_INTERNAL_OPTIONS
+/* default values for internal options are kept in a separate struct */
+#  define OPTION_COMMAND_INTERNAL(type, name, default_value, command_line_option, \
+                                  statement, description, flag, pcache) \
+{ command_line_option, offsetof(options_t, name), sizeof(type), \
+  OPTION_TYPE_##type, pcache, OPTION_MOD_##flag},
+
+#  define OPTION_COMMAND(type, name, default_value, command_line_option, \
+                         statement, description, flag, pcache) /* nothing */
+
+/* Read only source for default internal option values and names no lock needed
+ * since never written.
+ */
+static const option_t internal_option_traits[] = {
+#  include "optionsx.h"
```

Comment:
I know some of these were like this before but while we're changing these lines we may as well match the current style: single space after `#` since inside single level of #if, here and defines above.

Reply:
Fixed.